torch/fx/passes/_tensorify_python_scalars.py

```diff
@@ -249,28 +248,36 @@ def _sympy_interp(expr: sympy.Expr) -> MetaProxy:
                     node.replace_all_uses_with(replacement_proxy.node)
                     graph.erase_node(node)
 
-    # DCE symbols (which are guaranteed to be pure) only
-    graph.eliminate_dead_code()
-
-    # Now do one more pass that specializes all symfloats we didn't manage
-    # to tensorify away.
     nodes = list(graph.nodes)
-    for i, node in enumerate(nodes[:-1]):
+    for i, node in enumerate(nodes[-2::-1]):
         with graph.inserting_before(
-            nodes[i + 1] if node not in placeholders else first_non_placeholder
+            nodes[-(i + 2)] if node not in placeholders else first_non_placeholder
         ):
+            if (
+                len(node.users) == 0
+                and node.op != "placeholder"
+                and node.op != "output"
+                and not node.is_impure()
+            ):
+                graph.erase_node(node)
+                continue
+
             args = []
             kwargs: Dict[str, Any] = {}
             transform = False
             for a in node.args:
-                if a in specialized_float_nodes:
-                    args.append(specialized_float_nodes[a])
-                elif (
+                if (
                     isinstance(a, fx.Node)
                     and "val" in a.meta
                     and isinstance(zf := a.meta["val"], torch.SymFloat)
```

Comment:
I need to think about it more but I'm thinking about an edge case involving a float tensor 0d input that turns into a SymInt and then feeds into an operator...

Reply:
I updated the test suite to test `math.floor(math.sqrt(a.numel())),` where a is a 0d float tensor. LMK if you were thinking about something else.

Reply:
It's the reverse situation. Consider:

```
def f(x: float):
  x2 = some_untensorifyable_operation(x)
  y = math.floor(x2)
  return some_tensor_op(..., y, ...)
```

You need to specialize y, because it depends on a zf0 float after we tensorify inputs. But it's a SymInt, not a SymFloat, so the condition here fails to trigger.